dangerzone/isolation_provider/qubes.py

```diff
@@ -126,6 +138,12 @@ def _convert(
         text = "Converted document to pixels"
         self.print_progress_trusted(document, False, text, percentage)
 
+        if getattr(sys, "dangerzone_dev", False):
+            untrusted_log = read_debug_text(p)
```

Comment:
How about something like this commit "[Qubes: close qrexec stdin and stout](https://github.com/freedomofpress/dangerzone/pull/386/commits/d2ab6dbf2dccdc116b191fe08e30d9418cd72cb8)". In it I closed the pipe after the maximum permitted is reached. I used the opportunity to close of stdout as well after the last bit is received.

But also keep in mind that it's still possible for servers the keep the client hanging in other occasions. This will be solved with timeouts.

Reply:
> How about something like this commit "[Qubes: close qrexec stdin and stout](https://github.com/freedomofpress/dangerzone/pull/386/commits/d2ab6dbf2dccdc116b191fe08e30d9418cd72cb8)". In it I closed the pipe after the maximum permitted is reached. I used the opportunity to close of stdout as well after the last bit is received.

Looks good. I tested it out with the above code and the qube shuts down.

> But also keep in mind that it's still possible for servers the keep the client hanging in other occasions. This will be solved with timeouts.

Definitely, that's a different case.